### DISCUSSION TITLE
Separate routes and handlers for admin type

### DISCUSSION THREAD

**Discussion by RobertTutsek** [lexik/LexikJWTAuthenticationBundle#1059] [Q&A] [answered]
Hello,

As the title says Is it possible to create a separate login route with a separate provider (Admin) with this library?
I am not sure how I can achieve this.

Regards

**Comment by fd6130** [comment, marked as answer]
You can create a new controller with route for that case and generate the token manually using service from [docs](https://github.com/lexik/LexikJWTAuthenticationBundle/blob/2.x/Resources/doc/7-manual-token-creation.rst).

**Reply by RobertTutsek** [reply]
Hello,

Thank you for your reply, the solution which I went with spawned from the link you gave. So the solution for me was as follows:
- create an admin entity with repository
- create a provider for admin
- create a controller with /api/login-admin route
- JWTTokenManagerInterface $jwtManager ($this->jwtManager->create($admin))

Thank you for the help!

**Reply by RobertTutsek** [reply]
Hello again,

It works but now I cannot differentiate between Admin and User.
User generated tokens can access admin protected resources which is bad.

Any insights would be appreciated.

Regards.

**Reply by fd6130** [reply]
> It works but now I cannot differentiate between Admin and User. User generated tokens can access admin protected resources which is bad.

You can differentiate Admin and User by using user role [(symfony docs)](https://symfony.com/doc/current/security.html#roles).

Example: `ROLE_ADMIN` and `ROLE_USER`.

**Reply by RobertTutsek** [reply]
Hello,

I manage to find a solution.
The thing is I am using a different table for admins so roles don't help me in this case.

The solution was to add a chain provider (combining user and admin entities) and setting it to the api_admin and api firewalls, otherwise the incoming token was identified as either user or either admin and identified as admin even in case it was user.

Regards,
Thanks for the help!
